main/acle.md

```diff
@@ -11073,546 +11419,680 @@ Multi-vector floating-point round to integral value
+element types.
 
 
-  // Variants are also available for _s16_x4, _f16_x4 and _bf16_x4
-  void svstnt1[_u16_x4](svcount_t png, uint16_t *rn, svuint16x4_t zt)
-    __arm_streaming;
+``` c
+  // Variants are also available for _u8_x2, _u16_x2, _s16_x2, _f16_x2,
+  // _bf16_x2, _u32_x2, _s32_x2, _f32_x2, _u64_x2, _s64_x2 and _f64_x2
+  svint8x2_t svzipq[_s8_x2](svint8x2_t zn) __arm_streaming;
 
 
-  // Variants are also available for _s16_x2, _f16_x2 and _bf16_x2
-  void svstnt1_vnum[_u16_x2](svcount_t png, uint16_t *rn, int64_t vnum,
-                             svuint16x2_t zt)
-    __arm_streaming;
+  // Variants are also available for _u8_x4, _u16_x4, _s16_x4, _f16_x4,
+  // _bf16_x4, _u32_x4, _s32_x4, _f32_x4, _u64_x4, _s64_x4 and _f64_x4
+  svint8x4_t svzipq[_s8_x4](svint8x4_t zn) __arm_streaming;
+  ```
 
+#### UZP
 
-  // Variants are also available for _s16_x4, _f16_x4 and _bf16_x4
-  void svstnt1_vnum[_u16_x4](svcount_t png, uint16_t *rn, int64_t vnum,
-                             svuint16x4_t zt)
+Multi-vector unzip.
+
+``` c
+  // Variants are also available for _u8_x2, _u16_x2, _s16_x2, _f16_x2,
+  // _bf16_x2, _u32_x2, _s32_x2, _f32_x2, _u64_x2, _s64_x2 and _f64_x2
+  svint8x2_t svuzp[_s8_x2](svint8x2_t zn) __arm_streaming;
+
+
+  // Variants are also available for _u8_x4, _u16_x4, _s16_x4, _f16_x4,
+  // _bf16_x4, _u32_x4, _s32_x4, _f32_x4, _u64_x4, _s64_x4 and _f64_x4
+  svint8x4_t svuzp[_s8_x4](svint8x4_t zn) __arm_streaming;
+  ```
+
+The `svuzpq` intrinsics operate on quad-words, but for convenience accept all
+element types.
+
+``` c
+  // Variants are also available for _u8_x2, _u16_x2, _s16_x2, _f16_x2,
+  // _bf16_x2, _u32_x2, _s32_x2, _f32_x2, _u64_x2, _s64_x2 and _f64_x2
+  svint8x2_t svuzpq[_s8_x2](svint8x2_t zn) __arm_streaming;
+
+
+  // Variants are also available for _u8_x4, _u16_x4, _s16_x4, _f16_x4,
+  // _bf16_x4, _u32_x4, _s32_x4, _f32_x4, _u64_x4, _s64_x4 and _f64_x4
+  svint8x4_t svuzpq[_s8_x4](svint8x4_t zn) __arm_streaming;
+  ```
+
+### Streaming-compatible versions of standard routines
+
+ACLE provides the following streaming-compatible functions,
+with the same behavior as the standard C functions that they
+are named after. All of the functions have external linkage.
+
+``` c
+  void *__arm_sc_memcpy(void *dest, const void *src, size_t n)
+    __arm_streaming_compatible;
+
+  void *__arm_sc_memmove(void *dest, const void *src, size_t n)
+    __arm_streaming_compatible;
+
+  void *__arm_sc_memset(void *s, int c, size_t n)
+    __arm_streaming_compatible;
+
+  void *__arm_sc_memchr(void *s, int c, size_t n)
+    __arm_streaming_compatible;
+```
+
+
+### SVE2.1 and SME2 instruction intrinsics
+
+The functions in this section are defined by either the header file
+ [`<arm_sve.h>`](#arm_sve.h) or [`<arm_sme.h>`](#arm_sme.h)
+when `__ARM_FEATURE_SVE2.1` or `__ARM_FEATURE_SME2` is defined, respectively.
+
+Because these instructions overlap between the SME2 and SVE2.1 when using in
```

Comment:
Yeah, tricky.  It's not really possible to express the restrictions directly using attributes.  How about something like:

These intrinsics can only be called from non-streaming code if `__ARM_FEATURE_SVE2p1` is defined.  They can only be called from streaming code if the appropriate SME feature macro is defined (see next paragraph).  They can only be called from streaming-compatible code if they could be called from both non-streaming code and streaming code.

And then drop the attributes from the function declarations.